tests/math_tests.cpp

```diff
@@ -2908,4 +2908,209 @@ TEST_CASE("reverse floor") {
+                {minilua::Value(i).with_origin(minilua::LiteralOrigin()),
+                 minilua::Value(j).with_origin(minilua::LiteralOrigin())});
+            ctx = ctx.make_new(list);
+            auto res = minilua::math::fmod(ctx);
+            REQUIRE(res == minilua::Value(-2.5));
+
+            auto result = res.force(1);
+            REQUIRE(result.has_value());
+
+            CHECK(
+                result.value().collect_first_alternative()[0] ==
+                minilua::SourceChange(minilua::Range(), "5.2"));
+        }
+    }
+
+    SECTION("invalid force") {
+        double i = -2.5;
+        double j = 4.2;
+        minilua::Vallist list = minilua::Vallist(
+            {minilua::Value(i).with_origin(minilua::LiteralOrigin()),
+             minilua::Value(j).with_origin(minilua::LiteralOrigin())});
+        ctx = ctx.make_new(list);
+        auto res = minilua::math::fmod(ctx);
+        REQUIRE(res == minilua::Value(-2.5));
+
+        auto result = res.force("1");
+        CHECK_FALSE(result.has_value());
+
+        result = res.force(5);
+        CHECK_FALSE(result.has_value());
+    }
+}
+
+TEST_CASE("reverse log") {
+    minilua::Environment env;
+    minilua::CallContext ctx(&env);
+
+    SECTION("valid force") {
+        int x = 3;
+        minilua::Vallist list = minilua::Vallist(
+            {minilua::Value(x).with_origin(minilua::LiteralOrigin()), minilua::Nil()});
+        ctx = ctx.make_new(list);
+        auto res = minilua::math::log(ctx);
+        auto n = std::get<minilua::Number>(res);
+        REQUIRE(n.value == Approx(1.0986122886681));
+
+        auto result = res.force(0);
+        REQUIRE(result.has_value());
+
+        CHECK(
+            result.value().collect_first_alternative()[0] ==
+            minilua::SourceChange(minilua::Range(), "1"));
+    }
+
+    SECTION("invalid force") {
+        SECTION("try to force a invalid value") {
+            int x = 3;
+            minilua::Vallist list = minilua::Vallist(
+                {minilua::Value(x).with_origin(minilua::LiteralOrigin()), minilua::Nil()});
+            ctx = ctx.make_new(list);
+            auto res = minilua::math::log(ctx);
+            auto n = std::get<minilua::Number>(res);
+            REQUIRE(n.value == Approx(1.0986122886681));
+
+            auto result = res.force("1");
+            CHECK_FALSE(result.has_value());
+        }
+
+        SECTION("valid force, but log was called with 2 arguments") {
+            int x = 3;
+            int y = 2;
+            minilua::Vallist list = minilua::Vallist(
+                {minilua::Value(x).with_origin(minilua::LiteralOrigin()),
+                 minilua::Value(y).with_origin(minilua::LiteralOrigin())});
+            ctx = ctx.make_new(list);
+            auto res = minilua::math::log(ctx);
+            auto n = std::get<minilua::Number>(res);
+            REQUIRE(n.value == Approx(1.5849625007212));
+
+            auto result = res.force(0);
+            CHECK_FALSE(result.has_value());
```

Comment:
Just a nitpick but I think it is easier to read if you write `CHECK(!result.has_value())` instead of using `CHECK_FALSE`.

Reply:
For me its more obvious with `CHECK_FALSE` what value is expected.